### PR TITLE
[CORE-540] adding the context filter to the custom field listing

### PR DESCRIPTION
**Pull request by paultimoce**
The context filter on custom field listing API endpoint was released some time ago but was not yet documented.

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -518,6 +518,28 @@ Get a list of all the definitions of custom fields.
...
+            + context (enum[string], optional)
+                + Members
+                    + contact
+                    + company
+                    + form
+                    + event
+                    + product
+                    + project
+                    + milestone
+                    + sale
```
Comment (mrtus) [inline apiary.apib]: B: This one should be `deal`

Some other fields are not correct either.

**Review by mrtus — CHANGES_REQUESTED** [review]
I think we should fix our implementation first before releasing this. Half of these context fields are not correct or up to date.

/cc @stivni

File: apiary.apib (hunk 1 above)
Comment (paultimoce) [reply apiary.apib]: @mrtus - This is actually correct like is imo. Please check the for column in the crm_custom_fields table to see why

Comment (paultimoce) [reply apiary.apib]: You can also take a look at the CustomFieldDefinitionContext value object in core

File: apiary.apib [hunk 2]
```diff
@@ -518,6 +518,28 @@ Get a list of all the definitions of custom fields.
 
     + Attributes (object)
         + page (Page, optional)
+        + filter (object, optional)
+            + context (enum[string], optional)
+                + Members
+                    + contact
+                    + company
+                    + form
```
Comment (stivni) [inline apiary.apib]: B : this is not an existing type as far as the API is concerned
this is deprecated and we don't want it in the new API

File: apiary.apib [hunk 3]
```diff
@@ -518,6 +518,28 @@ Get a list of all the definitions of custom fields.
...
+                    + event
```
Comment (stivni) [inline apiary.apib]: B : this is not an existing type as far as the API is concerned
this is deprecated and we don't want it in the new API

File: apiary.apib [hunk 4]
```diff
@@ -518,6 +518,28 @@ Get a list of all the definitions of custom fields.
...
+                    + product
```
Comment (stivni) [inline apiary.apib]: PP : this is not an existing type as far as the API is concerned
we still need to add endpoints for products, so it doesn't make sense to have the custom fields for it

File: apiary.apib [hunk 5]
```diff
@@ -518,6 +518,28 @@ Get a list of all the definitions of custom fields.
...
+                    + project
+                    + milestone
+                    + sale
+                    + project_thirdparty
```
Comment (stivni) [inline apiary.apib]: B : this is not an existing type as far as the API is concerned
we are deprecating third parties in this form

File: apiary.apib [hunk 6]
```diff
@@ -518,6 +518,28 @@ Get a list of all the definitions of custom fields.
...
+                    + invoice
+                    + inv_invoice_item
```
Comment (stivni) [inline apiary.apib]: B : do we even want custom fields on invoice items?

File: apiary.apib [hunk 7]
```diff
@@ -518,6 +518,28 @@ Get a list of all the definitions of custom fields.
...
+                    + subscription
+                    + meeting_report
```
Comment (stivni) [inline apiary.apib]: B : is not part of the API yet

**Comment by paultimoce** [comment]
As discussed with @stivni, I added filtering documentation for the contexts that are already documented in the listing API endpoint

Will follow-up with a PR which will fix the implementation

**Review by WouterSioen — APPROVED** [review]

**Review by JordyDeGroeve — APPROVED** [review]

**Review by mrtus — CHANGES_REQUESTED** [review]
Good changes, but blocking until the implementation is fixed.

**Review by tiagobastos — APPROVED** [review]

**Review by dietrichm — APPROVED** [review]

**Review by jenssegers — APPROVED** [review]

**Review by ketlane — APPROVED** [review]

**Comment by marijnvandevoorde** [comment]
@mrtus can you unblock? Tnx!

**Comment by WouterSioen** [comment]
@mrnnn this should not be unblocked yet, since the implementation is not fixed yet. (We did a TA though about it and will fix this quite soon)

**Comment by thomasnoe07** [comment]
@dietrichm I see that this PR is still open although it is mentioned it will be fixed soon. Can you have a look if we can push this PR or not? tx

**Comment by stivni** [comment]
👉 poking @dietrichm & @WouterSioen to see if we can still do something with this...

**Comment by dietrichm** [comment]
This is going to be picked up within a new story in our backlog. 💪

**Pull request closed without merging**